File: Prueba_MVVM_java/app/src/main/java/com/example/prueba_mvvm_java/ui/gallery/GalleryViewModel.java
package com.example.prueba_mvvm_java.ui.gallery;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

import java.util.ArrayList;

public class GalleryViewModel extends ViewModel {

    private final MutableLiveData<String> mText;

    private MutableLiveData<ArrayList<String>> mNombres;

    public GalleryViewModel() {

        mText = new MutableLiveData<>();
        mText.setValue("This is gallery fragment");

        mNombres = new MutableLiveData<>();
        mNombres.setValue(new ArrayList<String>());


    }

    public LiveData<String> getText() {
        return mText;
    }

    public LiveData<ArrayList<String>> getNombres() {
        return mNombres;
    }

    public void addNombre(String nombre) {

        ArrayList<String> nombres = mNombres.getValue();

        nombres.add(nombre);

    }
}

File: ViewModelNavDrawer/app/src/main/java/com/example/viewmodelnavdrawer/ui/gallery/GalleryViewModel.kt
package com.example.viewmodelnavdrawer.ui.gallery

import android.app.Application
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel
import com.example.viewmodelnavdrawer.modelo.bd.AppBD
import com.example.viewmodelnavdrawer.modelo.data.Alumno

class GalleryViewModel() : ViewModel() {

    // TODO: Implement the ViewModel
    var mText: MutableLiveData<String>?
    var mNombres: MutableLiveData<ArrayList<String>>?
    var alumno: MutableLiveData<Alumno>?
    var mAlumnos: MutableLiveData<ArrayList<Alumno>>?
    var daoAlumno = AppBD.getAppDB(Application().applicationContext)!!.daoAlumno()

    init {
        this.mText = MutableLiveData<String>("Este es el fragmento main")
        this.mNombres = MutableLiveData(ArrayList<String>())
        this.alumno = null
        this.mAlumnos = daoAlumno.ver() as MutableLiveData<ArrayList<Alumno>>
    }

    fun addNombre(nombre: String) {
        val nombres: ArrayList<String>? = mNombres?.value
        nombres?.add(nombre)
    }

    fun addAlumno(alumno: Alumno) {
        val alumnos = mAlumnos?.value
        daoAlumno.crear(alumno)
        mAlumnos!!.postValue(alumnos)
    }
}